pyquil/operator_estimation.py

```diff
@@ -479,3 +541,93 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
+
+    :param list_of_pauli_terms: List of Pauli terms to check
+    :return: The highest weight Pauli Term
+    """
+    # create an unsorted (frozen)set of tuples of the form (qubit, operation)
+    unsorted_qubit_ops = reduce(lambda x, y: x | y, (pt.operations_as_set() for pt in list_of_pauli_terms))
+    # create a list of tuples of the form (qubit, operation), sorted by qubits
+    qubit_ops = sorted(list(unsorted_qubit_ops), key=lambda x: x[0])
+    # return a PauliTerm created from a list of tuples of the form (operation, qubit)
+    return PauliTerm.from_list([(i[1], i[0]) for i in qubit_ops])
+
+
+def _max_tpb_overlap(expt_setting: ExperimentSetting, diagonal_sets: Dict):
+    """
+    Calculate the max overlap of an ExperimentSetting with other ExperimentSettings specified
+    by keys of diagonal_sets.
+    Returns a different key if ``expt_setting`` isn't diagonal in the same tpb as any of the keys.
+    If it is diagonal in the same tpb as some key, then the corresponding value (a list of ExperimentSettings)
+    is updated to include ``expt_setting`` and the key is updated so it has the largest weight PauliTerm
+    specifying the tpb.
+
+    :param expt_setting: ExperimentSetting
+    :param diagonal_sets: dictionary keyed with ExperimentSetting, and with each value being a
+            list of ExperimentSettings
+    :return: the updated diagonal_sets dictionary updated with the input expt_setting placed
+            appropriately into it
+    """
+    # loop through dict items
+    for key, es_list in diagonal_sets.items():
+        # determine if key is diagonal in the same tpb as expt_setting
+        b_diagonal_tpb = _expt_settings_diagonal_in_tpb(key, expt_setting)
+        # update the dict value if so
+        if b_diagonal_tpb:
+            # determine the updated list of ExperimentSettings
+            updated_es_list = es_list + [expt_setting]
+            # obtain the diagonalizing bases for both the updated in and out sets
+            diagonalizing_in_term = _get_diagonalizing_basis([es.in_operator for es in updated_es_list])
+            diagonalizing_out_term = _get_diagonalizing_basis([es.out_operator for es in updated_es_list])
+            # update the diagonalizing basis (key of dict) if necessary
+            if len(diagonalizing_in_term) > len(key.in_operator) or len(diagonalizing_out_term) > len(key.out_operator):
+                del diagonal_sets[key]
+                new_key = ExperimentSetting(diagonalizing_in_term, diagonalizing_out_term)
+                diagonal_sets[new_key] = updated_es_list
+            else:
+                diagonal_sets[key] = updated_es_list
+            return diagonal_sets
+
+    # made it through entire dict without finding any ExperimentSetting with shared tpb,
+    # so need to make a new item
+    diagonal_sets[expt_setting] = [expt_setting]
+
+    return diagonal_sets
+
+
+def _commuting_sets_by_zbasis_tomo_expt(tomo_expt: TomographyExperiment):
```

Comment:
what does this function name mean

Reply:
Renamed this function to `_max_tpb_overlap(..)`, and moved the other function formerly named so into this one.